tests/unit/browser/test_adblock.py

```diff
@@ -0,0 +1,434 @@
+                         'one_per_line'  -->  one host per line format
+                         'not_correct'  -->  Not a correct hosts file format.
+    """
+
+    blocklist = QUrl(os.path.join(str(directory), name))
+    with open(blocklist.path(), 'w', encoding='UTF-8') as hosts:
+        # ensure comments are ignored when processing blocklist
+        hosts.write('# Blocked Hosts List #\n\n')
+        if line_format == 'etc_hosts': # /etc/hosts like format
+            for path in blocked_hosts:
+                hosts.write('127.0.0.1  ' + path + '\n')
+        elif line_format == 'one_per_line':
+            for path in blocked_hosts:
+                hosts.write(path + '\n')
+        elif line_format == 'not_correct':
+            for path in blocked_hosts:
+                hosts.write(path + ' This is not a correct hosts file\n')
+    return blocklist
+
+
+def assert_urls(host_blocker,
+                blocked=BLOCKLIST_HOSTS,
+                whitelisted_hosts=WHITELISTED_HOSTS,
+                urls_to_check=URLS_TO_CHECK):
+
+    """Test if urls_to_check are effectively blocked or not by HostBlocker
+       Ensure Urls in blocked and not in whitelisted are blocked
+       All other Urls are not to be blocked."""
+
+    whitelisted = list(whitelisted_hosts) + list(host_blocker.WHITELISTED)
+    for str_url in urls_to_check:
+        url = QUrl(str_url)
+        host = url.host()
+        if host in blocked and host not in whitelisted:
+            assert host_blocker.is_blocked(url)
+        else:
+            assert not host_blocker.is_blocked(url)
+
+
+def generic_blocklists(directory):
+
+    """Return a generic list of files to be used in hosts-block-lists option.
+       This list contains :
+       - a remote zip file with 1 hosts file and 2 useless files
+       - a remote zip file with only useless files
+           (Should raise a FileNotFoundError)
+       - a remote zip file with only one valid hosts file
+       - a local text file with valid hosts
+       - a remote text file without valid hosts format."""
+
+    # remote zip file with 1 hosts file and 2 useless files
+    file1 = create_blocklist(directory,
+                             blocked_hosts=CLEAN_HOSTS,
+                             name='README',
+                             line_format='not_correct')
+    file2 = create_blocklist(directory,
+                             blocked_hosts=BLOCKLIST_HOSTS[:3],
+                             name='hosts',
+                             line_format='etc_hosts')
+    file3 = create_blocklist(directory,
+                             blocked_hosts=CLEAN_HOSTS,
+                             name='false_positive',
+                             line_format='one_per_line')
+    files_to_zip = [file1.path(), file2.path(), file3.path()]
+    zip_path = create_zipfile(directory, files_to_zip, 'block1')
+    blocklist1 = QUrl(zip_path)
+
+    # remote zip file without file named hosts
+    # (Should raise a FileNotFoundError)
```

Comment:
I don't think so - the exception is just handled. `adblock.guess_zip_filename()` is called by `get_fileobj()`, and that in turn is called by `_merge_file()`, which handles `OSError` and displays an error - and `FileNotFoundError` is a subclass of `OSError`, so it's handled there as well.